File: 181 excersises/160.py
import math

def es_cuadrado(x):
    if x < 0:
        return False
    s = math.isqrt(x)
    return s * s == x

def es_fibonacci(n):
    if n < 0:
        return False
    val = 5 * n**2
    return es_cuadrado(val + 4) or es_cuadrado(val - 4)

num = int(input("Ingresa un numero: "))
if es_fibonacci(num):
    print(f"{num} es un numero Fibonacci")
else:
    print(f"{num} NO es un numero Fibonacci")

File: 181 excersises/160.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool esCuadradoPerfecto(int x) {
    if (x < 0) return false;
    int s = sqrt(x);
    return (s * s == x);
}

bool esFibonacci(int n) {

    if (n < 0) return false;
    int val = 5 * n * n;
    return esCuadradoPerfecto(val + 4) || esCuadradoPerfecto(val - 4);
}

int main() {
    int num;
    cout << "Ingrese un número: ";
    cin >> num;
    
    if (esFibonacci(num)) {
        cout << num << " es un número Fibonacci" << endl;
    } else {
        cout << num << " NO es un número Fibonacci" << endl;
    }
    
    return 0;
}